src/generator/AutoRest.Go/Model/MethodGo.cs

```diff
@@ -245,12 +245,16 @@ public List<string> PrepareDecorators
               );
         }
 
-        if (RequestHeaders.Any())
+        if (HeaderParameters.Any())
```

Comment:
Can you add a comment explaining why HeaderParameters instead of RequestHeaders?